File: Punteggiatura/src/Punteggiatura.java
import java.util.Scanner;

public class Punteggiatura {
	
	private static Scanner in = new Scanner(System.in);
	private static String str1;
	private static int somma = 0;
	private static String punti = ",.;:!?";
	
	public static void main(String[] args) {
		
		System.out.println("Inserisci le stringhe compresi i caratteri di punteggiatura: ");
		
		do{
			str1 = in.nextLine();
			if(str1.length() > 0)
			for(int i = 0; i < str1.length(); i++)
				for(int j = 0; j < punti.length(); j++)
					if(str1.charAt(i) == punti.charAt(j))
						somma++;
			
		}while(str1.length() > 0);
		System.out.println(somma);
	}
}

File: VEcchi/Python/Punteggiatura.py
str1 = "prova"
punti = ",.;:!?"
somma = 0

while len(str1) > 0:
    str1 = raw_input("Inserisci stringhe contenenti caratteri di punteggiatura: ")
    if len(str1) >0:
        for i in range(0, len(str1)):
            for j in range(0, len(punti)):
                if str1[i] == punti[j]:
                    somma += 1
print somma
